### PR TITLE
Externalize KVM Agent storage's timeout configuration

### PR DESCRIPTION
**Pull request by GutoVeronezi**
### Description
On KVMHAMonitor, the heartbeat timeout is hardcoded to `60000` ms; therefore, operators do not have options to choose which timeout they want/need. 

This PR intends to externalize the heartbeat timeout configuration on KVMHAMonitor.

Depending on the use case, this timeout might need increasing/decreasing. Therefore, it is interesting to externalize such configurations for cloud operators. The hardcoded value is taken as the default one. Therefore, the current behavior is maintained.

### Types of changes
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functioanlity)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale
- [x] Major
- [ ] Minor

### How Has This Been Tested?
It has been tested locally on a test lab.
1. I changed my log4j to debug mode.
2. I added and changed `hearbeat.update.timeout` setting to `agent.properties` file several times.
3. I restarted agent and watched the log.

### CODE REVIEW

**Comment by GutoVeronezi** [comment]
@DaanHoogland @nvazquez as we discussed, I created a new PR proposing the reverted work. At the time you raised the issue (in https://github.com/apache/cloudstack/pull/4585#issuecomment-882565561), I did not have a chance to investigate and validate it. I understand the use of revert to get things working. However, recently I made some tests with the PR and `addHost`'s feature is working (you can see one [HERE](https://www.youtube.com/watch?v=0loAaOFOipU)). Therefore, it does not seem to be a problem with the PR itself; analyzing the logs, the issue you showed us seems related to the environment.

I think that next time something like this happens, we should investigate the real problem and pin point it in some part of the code, instead of just reverting the first thing we see in hopes that this fixes the issue.

**Comment by DaanHoogland** [comment]
> @DaanHoogland @nvazquez as we discussed, I created a new PR proposing the reverted work. At the time you raised the issue (in [#4585 (comment)](https://github.com/apache/cloudstack/pull/4585#issuecomment-882565561)),

great, I'll run the tests.

> I think that next time something like this happens, we should investigate the real problem and pin point it in some part of the code, instead of just reverting the first thing we see in hopes that this fixes the issue.

It was not hoping @GutoVeronezi , I tested the revert and the PR itself. I indeed had no time to get to the bottom of it. Also I do/did not believe for a moment it was in the PR itself but in some sort of conflict with other code.

**Comment by nvazquez** [comment]
Thanks @GutoVeronezi I'll review it. I think the problem was the PR was merged with outdated test results. After merging the PR every automated test env failed, which makes me think we could have detected the issue attempting a new round of tests. However, I do not agree on the last sentence, @DaanHoogland made sure reverting the changes made the main branch fuctional again before reverting

File: plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/KVMHAMonitor.java
Comment (nvazquez) [inline plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/KVMHAMonitor.java:155]: Minor suggestion: what about adding the host IP and a boolean parameter, and add them to `cmd` in case the host IP is not null or the boolean is true?

**Review by nvazquez — COMMENTED** [review]
Looks good, well documented and tests added, thanks @GutoVeronezi - just left a minor comment

File: plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/KVMHAMonitor.java
Comment (GutoVeronezi) [reply plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/KVMHAMonitor.java:155]: @nvazquez I think I did not get it :grimacing:, could you explain me better?

Comment (nvazquez) [reply plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/KVMHAMonitor.java:155]: Sure, I meant to include both parameters so the method's header becomes:
`private Script createHeartBeatCommand(NfsStoragePool primaryStoragePool, String hostIp, boolean cleanup)`

Then:
````
if (StringUtils.isNotBlank(hostIp) {
   cmd.add("-h", hostIp);
}
if (cleanup) {
   cmd.add("-c");
}
````

Comment (GutoVeronezi) [reply plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/KVMHAMonitor.java:155]: Understood, I will do it.

Comment (GutoVeronezi) [reply plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/KVMHAMonitor.java]: @nvazquez done. Before this changes, `hostPrivateIp` would be added independently if it is null or empty, so adopted your suggestion and made a little change to keep this behavior.

**Review by nvazquez — APPROVED** [review]
Code LGTM

File: plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/KVMHAMonitor.java
Comment (nvazquez) [reply plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/KVMHAMonitor.java]: Great, thanks @GutoVeronezi

**Review by RodrigoDLopez — APPROVED** [review]
Code LGTM.
Do you wanna some manual tests here?

**Comment by weizhouapache** [comment]
@blueorangutan test

**Comment by weizhouapache** [comment]
@blueorangutan test

File: agent/conf/agent.properties
Comment (weizhouapache) [inline agent/conf/agent.properties:272]: @GutoVeronezi 
please add a "\n" to the end of the file.
it caused failures in trillian test that fresh installed host cannot be added to zone.

in /etc/cloudstack/agent/agent.properties,
```
# heartbeat.update.timeout=60000keystore.passphrase=RFXZya8a4esacWaG
```
should be
```
# heartbeat.update.timeout=60000
keystore.passphrase=RFXZya8a4esacWaG
```

**Review by weizhouapache — CHANGES_REQUESTED** [review]
@nvazquez 
please do not merge this PR until my comment above is addreesed.

@GutoVeronezi 
I manually changed the original /etc/cloudstack/agent/agent.properties in trillian test, all tests passed.
```
[root@pr5239-t1382-kvm-centos7-kvm2 ~]# cp /etc/cloudstack/agent/agent.properties /etc/cloudstack/agent/agent.properties.orig
[root@pr5239-t1382-kvm-centos7-kvm2 ~]# sed -i "s/^# heartbeat.update.timeout=60000$/# heartbeat.update.timeout=60000\n/g" /etc/cloudstack/agent/agent.properties
[root@pr5239-t1382-kvm-centos7-kvm2 ~]# diff -u /etc/cloudstack/agent/agent.properties.orig /etc/cloudstack/agent/agent.properties
--- /etc/cloudstack/agent/agent.properties.orig	2021-07-26 20:03:11.365980320 +0000
+++ /etc/cloudstack/agent/agent.properties	2021-07-26 20:03:35.571938116 +0000
@@ -269,4 +269,4 @@
 
 # This parameter specifies the heartbeat update timeout in ms; The default value is 60000ms (1 min).
 # Depending on the use case, this timeout might need increasing/decreasing.
-# heartbeat.update.timeout=60000
\ No newline at end of file
+# heartbeat.update.timeout=60000
```

**Comment by nvazquez** [comment]
Great catch @weizhouapache, @GutoVeronezi please check the open comment

File: agent/conf/agent.properties
Comment (GutoVeronezi) [reply agent/conf/agent.properties]: @weizhouapache done, thanks!

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan test

**Review by weizhouapache — APPROVED** [review]
testing looks ok.

**Pull request merged**
